src/ProductConstructionService/ProductConstructionService.BarViz/Components/CodeflowContextMenu.razor

```diff
@@ -0,0 +1,210 @@
+    {
+        <FluentMenuItem>
+            <ChildContent>
+                Backflow
+                <span slot="start">
+                    <FluentIcon Value="@(new Icons.Regular.Size16.ArrowLeft())" Color="@Color.Neutral" />
+                </span>
+            </ChildContent>
+            <MenuItems>
+                <FluentMenuItem OnClick="@(() => ShowDetails(Codeflow.BackflowSubscription))">
+                    Show subscription
+                    <span slot="start">
+                        <FluentIcon Value="@(new Icons.Regular.Size16.DocumentOnePage())" Color="@Color.Neutral" />
+                    </span>
+                </FluentMenuItem>
+
+                <FluentMenuItem OnClick="@(() => Clipboard.SetTextAsync(Codeflow.BackflowSubscription.Id.ToString()))">
+                    Copy subscription ID
+                    <span slot="start">
+                        <FluentIcon Value="@(new Icons.Regular.Size16.Copy())" Color="@Color.Neutral" />
+                    </span>
+                </FluentMenuItem>
+
+                @if (Codeflow.BackflowSubscription.Enabled)
+                {
+                    <FluentMenuItem OnClick="@TriggerBackflow">
+                        Trigger subscription
+                        <span slot="start">
+                            <FluentIcon Value="@(new Icons.Regular.Size16.Flash())" Color="@Color.Neutral" />
+                        </span>
+                    </FluentMenuItem>
+                }
+
+                @if (Codeflow.BackflowSubscription?.LastAppliedBuild != null)
+                {
+                    <FluentMenuItem OnClick="@(async () => await JSRuntime.OpenNewWindow(BackflowBuildLink!))">
+                        Show last flown VMR build
+                        <span slot="start">
+                            <FluentIcon Value="@(new Icons.Regular.Size16.Flashlight())" Color="@Color.Neutral" />
+                        </span>
+                    </FluentMenuItem>
+
+                    <FluentMenuItem OnClick="@(async () => await JSRuntime.OpenNewWindow(Codeflow.BackflowSubscription.LastAppliedBuild.GetBuildLink()))">
+                        Go to build
+                        <span slot="start">
+                            <FluentIcon Value="@(new Icons.Regular.Size16.CloudLink())" Color="@Color.Neutral" />
+                        </span>
+                    </FluentMenuItem>
+
+                    <FluentMenuItem OnClick="@(async () => await JSRuntime.OpenNewWindow(Codeflow.BackflowSubscription.LastAppliedBuild.GetCommitLink()))">
+                        Go to commit
+                        <span slot="start">
+                            <FluentIcon Value="@(new Icons.Regular.Size16.DocumentLink())" Color="@Color.Neutral" />
+                        </span>
+                    </FluentMenuItem>
+                }
+            </MenuItems>
+        </FluentMenuItem>
+    }
+
+    @if (Codeflow.ForwardflowSubscription != null)
```

Comment:
Both ForwardFlow and BackFlow sections feel pretty similiar, possibly similar enough to be a component that takes a `CodeflowSubscription` and renders the `FluentMenuItem`

Reply:
Good point, will change

Reply:
Done